routers/api/v1/repo/pull.go

```diff
@@ -508,6 +510,30 @@ func EditPullRequest(ctx *context.APIContext, form api.EditPullRequestOption) {
 		notification.NotifyIssueChangeStatus(ctx.User, issue, statusChangeComment, issue.IsClosed)
 	}
 
+	// change pull target branch
+	if len(form.Base) != 0 && form.Base != pr.BaseBranch {
+		if !ctx.Repo.GitRepo.IsBranchExist(form.Base) {
```

Comment:
If we're going to error out we should do it before we make any changes. (This is not your fault - the same problem occurs throughout this function.) But it's kinda crazy that we could change all of the other settings, return an error and still have those changes persisting.)